### PR TITLE
Add conversions for Byte and Char to JsonDSL

### PR DESCRIPTION
**Pull request by ponkotuy**

### CODE REVIEW

**Comment by dozed** [comment]
Could you add another test case which tests conversion of `List[_]` for `Double`, `Char` and `Byte` to `JValue`?

**Comment by ponkotuy** [comment]
I see. I'll do it.

**Comment by ponkotuy** [comment]
Because I'm not aware of conversion of List[_] for Double, I didn't.

File: tests/src/test/scala/org/json4s/JsonDSLSpec.scala [hunk 1]
```diff
@@ -26,10 +26,24 @@ object JsonDSLSpec extends Specification with JValueGen with ScalaCheck {
...
-      val buildProp = (shortValue: Short) =>
-        (("shortValue" -> shortValue) ~ ("dummy", "dummy")) must_== JObject(("shortValue", JInt(shortValue)), ("dummy", JString("dummy")))
+      val buildProp = (shortValue: Short, byteValue: Byte, charValue: Char) => {
+        val dsl = (("shortValue" -> shortValue) ~
+          ("byteValue" -> byteValue) ~
+          ("charValue" -> charValue) ~
+          ("shortArray" -> List[Short](shortValue)) ~
+          ("byteArray" -> List[Byte](byteValue)) ~
+          ("charArray" -> List[Char](charValue)))
+        val correct = JObject(
```
Comment (seratch) [inline tests/src/test/scala/org/json4s/JsonDSLSpec.scala]: expected?

**Comment by seratch** [comment]
LGTM

**Comment by dozed** [comment]
Thanks!

**Pull request merged**